vcstools/prof_utils.py

```diff
@@ -555,6 +578,470 @@ def check_clip(prof_to_check, toomuch=0.8, toolittle_frac=0., toolittle_absolute
+
+    overest_on_pulse = []
+    underest_on_pulse = []
+    for root in real_max:
+        # Underestimated on-pulse
+        # Find where the index of root would be
+        dy2_roots = np.append(dy2_roots, root)
+        dy2_roots = sorted(dy2_roots)
+        i = dy2_roots.index(root)
+        dy2_roots.remove(root)
+        try:
+            underestimate = [round(dy2_roots[i-1]), round(dy2_roots[i])]
+        except IndexError as e:
+            # The next inflection is on the other side of the profile
+            if i == 0:  # Left side
+                underestimate = [round(dy2_roots[-1]), round(dy2_roots[i])]
+            else:  # Right side
+                underestimate = [round(dy2_roots[i-1]), round(dy2_roots[0])]
+        underest_on_pulse.append(underestimate)
+        
+        # Overestimated on-pulse
+        false_max = np.append(false_max, root)
+        false_max = sorted(false_max)
+        j = false_max.index(root)
+        false_max.remove(root)
+        # Append the overestimate
+        try:
+            overestimate = [round(false_max[j-1]), round(false_max[j])]
+        except IndexError as e:
+            # The next inflection is on the other side of the profile
+            if j == 0:  # Left side
+                overestimate = [round(false_max[-1]), round(false_max[j])]
+            else:  # Right side
+                overestimate = [round(false_max[j-1]), round(false_max[0])]
+        # If the bounds are touching
+        if overest_on_pulse and overestimate[0] == overest_on_pulse[-1][1]:
+            overest_on_pulse[-1][1] = overestimate[1]
+        # If the last element wraps around to the first
+        elif overest_on_pulse and overestimate[1] == overest_on_pulse[0][0]:
+            overest_on_pulse[0][0] = overestimate[0]
+        else:
+            overest_on_pulse.append(overestimate)
+    
+    # Remove any duplicate items because somehow this happens sometimes
+    overest_on_pulse.sort()
+    overest_on_pulse = list(k for k, _ in itertools.groupby(overest_on_pulse))
+    underest_on_pulse.sort()
+    underest_on_pulse = list(
+        k for k, _ in itertools.groupby(underest_on_pulse))
+
+    # Fill the on-pulse regions
+    overest_on_pulse = _fill_on_pulse_region(
+        overest_on_pulse, splined_profile, norm_prof, clip_noise)
+    underest_on_pulse = _fill_on_pulse_region(
+        underest_on_pulse, splined_profile, norm_prof, clip_noise)
+    # Get the off-pulse pairs as well
+    overest_off_pulse = get_off_pulse(overest_on_pulse)
+    underest_off_pulse = get_off_pulse(underest_on_pulse)
+
+    # Work out the noise std and mean using the oversetimated on-pulse region
+    off_pulse_noise_std = np.nanstd(
+        profile_region_from_pairs(norm_prof, overest_off_pulse))
+    off_pulse_noise_mean = np.nanmean(
+        profile_region_from_pairs(normamlise_prof(profile), overest_off_pulse))
```

Comment:
Is there a reason both of these aren't norm_prof (instead of normamlise_prof(profile))

Reply:
yes, "norm_prof" has had its first estimate noise mean removed. so the off_pulse_noise_mean would be close to zero if i used it there